scrapy/core/downloader/__init__.py

```diff
@@ -143,8 +165,42 @@ def _deactivate(response: _T) -> _T:
         )
         return dfd.addBoth(_deactivate)
 
+    def _count_backout(self, reason):
+        self._stats.inc_value("request_backouts/total")
+        self._stats.inc_value(f"request_backouts/{reason}")
+
     def needs_backout(self) -> bool:
-        return len(self.active) >= self.total_concurrency
+        if len(self.active) >= self.total_concurrency:
+            self._count_backout("concurrent_requests")
+            return True
+        if (
+            self._response_max_active_size
+            and self.middleware.response_active_size >= self._response_max_active_size
+        ):
+            if not self._response_max_active_size_warned:
+                self._response_max_active_size_warned = True
+                logger.info(
+                    f"The active response size, i.e. the total size of all "
+                    f"bodies from responses that have been processed by "
+                    f"downloader middlewares and remain in memory, is "
+                    f"{self.middleware.response_active_size} B. The "
+                    f"RESPONSE_MAX_ACTIVE_SIZE setting sets its maximum value "
+                    f"at {self._response_max_active_size} B. No more requests "
+                    f"will be processed until active response size lowers. If "
+                    f"your memory allows it, you may increase "
+                    f"RESPONSE_MAX_ACTIVE_SIZE, which should increase your "
+                    f"crawl speed. If your code keeps non-weak references to "
+                    f"Response objects, your crawl might get stuck "
+                    f"indefinitely; you can set RESPONSE_MAX_ACTIVE_SIZE to 0 "
+                    f"to disable this limit, but then your code might run out "
+                    f"of memory. This message will only appear the first time "
+                    f"this happens. To learn how often request processing has "
+                    f"been paused during a crawl for this reason, see the "
+                    f"request_backouts/response_max_active_size stat."
+                )
+            self._count_backout("response_max_active_size")
```

Comment:
As You can see on log output from code sample from https://github.com/scrapy/scrapy/issues/6206#issuecomment-2180631979 (printing scraper slot active size every time when it's checked)
This type of.. tracking (by incrementing specific stats values +1 on each time when specific code lines reached) - will not be informative as engine checks backout conditions of scrapy components irregularly.

In this case it is preferred to track... time (in seconds) for what backout condition (scraper, downloader on something else) triggered and for how many seconds sending next requests to downloader was delayed.
With something like 





    clsss backout_timer:
       def start_or_continue_timer...
           # start time
       def drop_timer..
           ##add measured seconds to related crawler.stats values 
    ...
    def _needs_backout(self) -> bool:
        assert self.slot is not None  # typing
        assert self.scraper.slot is not None  # typing
        result = False
        if not self.running:
            backout_timed.drop_timer()
            return True
        if self.downloader.needs_backout():
            backout_timer.start_or_continue_timer('downloader')
            return True
        if self.scraper.slot.needs_backout()
            backout_timer.start_or_continue_timer('scraper')
            return True
        backout_timed.drop_timer()
        return False

Reply:
[Done](https://github.com/scrapy/scrapy/pull/6430/commits/36cf156500c0f421546fdb9df97b99db16300e35).

Reply:
I didn't expect fast feedback for this.
It 'd be good to make it possible to.. look into this on per minute basis from for example periodic_logstats extension.
If periodic_logstats extension used - it will show all backouts data except... active (because active didn't reflect on stats as it didn't finished). 
It may looks like:.. drop(write) backout timer and make another (the same reason) timer - called every 60 seconds interval (right before - periodic_log stats extension method) to get relevant backout data on per minute basis. 
Because we often have cases when.. some backout delay started and didn't finished.